src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor.cs

```diff
@@ -75,6 +68,53 @@ protected AbstractSymbolDisplayVisitor NotFirstVisitorNamespaceOrType
             }
         }
 
+        protected virtual void Allocate(
+            ArrayBuilder<SymbolDisplayPart> builder,
+            SymbolDisplayFormat format,
+            bool isFirstSymbolVisited,
+            SemanticModel? semanticModelOpt,
+            int positionOpt,
+            bool inNamespaceOrType)
+        {
+            Debug.Assert(format != null);
```

Comment:
is there value in teh assert, if we're just going to throw a moment later?

Reply:
⏱️  The throw is for the field, and catches cases where Allocate is called on an instance which was previously taken from the pool and not yet returned. I'll add a comment.

Reply:
➡️ Fixed